pkg/authenticator/authenticator.go

```diff
@@ -254,6 +264,35 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 	return nil
 }
 
+// We define these functions so we can mock WaitForInjectedCertificateFile
+type OsStatFunc func(name string) (os.FileInfo, error)
+type OsIsNotExistFunc func(err error) bool
+
+// WaitForInjectedCertificateFile waits for retryCountLimit seconds to see if the cert file
+// exists in the given path. If it's not there by the end of the retry count limit, it returns
+// an error.
+func (auth *Authenticator) WaitForInjectedCertificateFile(certificatePath string, retryCountLimit int, osStatFunc OsStatFunc, osIsNotExistFunc OsIsNotExistFunc) error {
```

Comment:
As was briefly discussed in slack, this probably seems appropriate to extract to its own file

Reply:
Line-too-long

Reply:
why would you move this to another file? would it be a simple util?

Reply:
yeah simple util. the problem is that this file is after this PR at 350+ lines and getting hard to untangle.

Reply:
thanks, will do.